client/app/components/queries/visualization-embed.js

```diff
@@ -37,7 +37,7 @@ export default function init(ngModule) {
     return session($http, $route, Auth).then(() => {
       const queryId = $route.current.params.queryId;
       const query = $http.get(`api/queries/${queryId}`).then(response => response.data);
-      const queryResult = $http.get(`api/queries/${queryId}/results.json${location.search}`).then(response => response.data);
+      const queryResult = $http.post(`api/queries/${queryId}/results${location.search}`).then(response => response.data);
```

Comment:
It looks like semantic flaw: HTTP POST should be used to update something, and GET - to retrieve data. But here you use POST to fetch query results.

Reply:
Well, it has a side-effect - it modifies system state by creating a new record in the query_results table, so you can look at it as updating something.

Reply:
Then makes sense. But looks weird anyway :slightly_smiling_face:

Reply:
Why are we passing parameters in the query string and not in the request body?

Reply:
[No apparent reason](https://github.com/getredash/redash/pull/3495/commits/818fc27bf028992d488156fdf1178e6a6e915a78).